react/training-project/src/app/screens/PrimaryLayout/index.js

```diff
@@ -1,40 +1,39 @@
 import React, { Component, Fragment } from 'react';
-import { Route, Switch, Redirect } from 'react-router-dom';
+import { Route, Switch } from 'react-router-dom';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 
-import actionCreators from '~redux/Auth/actions'; // eslint-disable-line
-import { GAME, PROFILE, LOGIN } from '~components/Routes/constants';
+/* eslint-disable import/order */
+import actionCreators from '~redux/Auth/actions';
+import { GAME, PROFILE } from '~components/Routes/constants';
+import Topbar from '~components/Topbar';
 
-import Topbar from './components/Topbar';
 import Game from './screens/Game';
 import Profile from './screens/Profile';
 
-class AuthorizedRoute extends Component {
+class PrimaryLayout extends Component {
   handleLogOut = () => this.props.logOut(this.props.token);
 
   render() {
-    return !this.props.token ? (
-      <Redirect to={LOGIN} />
-    ) : (
+    return (
       <Fragment>
         <Topbar logOut={this.handleLogOut} />
         <Switch>
           <Route path={GAME} component={Game} />
```

Comment:
This component looks very similar to the one in `app/index.js`.
The structure we use in Wolox should be:
- The App component (app/index.js) should call the Routes component.
- The Routes component will have the ConnectedRouter component, with the switch and the AuthorizedRoute.
- You can have a HOC called PrimaryLayout that may contain the header and the footer, and use it en every view that needs it, because it may happen that not all the views have them.